pkg/providers/amifamily/ami.go

```diff
@@ -94,45 +98,91 @@ func (p *Provider) KubeServerVersion(ctx context.Context) (string, error) {
 	return version, nil
 }
 
-// Get returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
+// MapInstanceTypes returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
 // If AMI overrides are specified in the AWSNodeTemplate, then only those AMIs will be chosen.
-func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, amiFamily AMIFamily) (map[string][]*cloudprovider.InstanceType, error) {
-	kubernetesVersion, err := p.KubeServerVersion(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("getting kubernetes version %w", err)
-	}
+func MapInstanceTypes(amis []AMI, instanceTypes []*cloudprovider.InstanceType) map[string][]*cloudprovider.InstanceType {
 	amiIDs := map[string][]*cloudprovider.InstanceType{}
-	amiRequirements, err := p.getAMIRequirements(ctx, nodeTemplate)
-	if err != nil {
-		return nil, err
-	}
-	if len(amiRequirements) > 0 {
-		// Iterate through AMIs in order of creation date to use latest AMI
-		amis := sortAMIsByCreationDate(amiRequirements)
-		for _, instanceType := range instanceTypes {
-			for _, ami := range amis {
-				if err := instanceType.Requirements.Compatible(amiRequirements[ami]); err == nil {
+
+	for _, instanceType := range instanceTypes {
+		for _, ami := range amis {
+			if !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceAWSNeuron]) {
+				if err := ami.Requirements.Compatible(
+					scheduling.NewRequirements(
+						scheduling.NewRequirement(v1alpha1.LabelInstanceGPUManufacturer, v1.NodeSelectorOpIn, v1alpha1.NVIDIAGPU, v1alpha1.AWSNeuron)),
+				); err == nil {
 					amiIDs[ami.AmiID] = append(amiIDs[ami.AmiID], instanceType)
 					break
 				}
+			} else if err := instanceType.Requirements.Compatible(ami.Requirements); err == nil {
+				amiIDs[ami.AmiID] = append(amiIDs[ami.AmiID], instanceType)
+				break
 			}
+
 		}
-		if len(amiIDs) == 0 {
-			return nil, fmt.Errorf("no instance types satisfy requirements of amis %v,", lo.Keys(amiRequirements))
+	}
+	return amiIDs
+}
+
+// Returning a list of AMIs with its associated requirements
+// the AMIs are sorted by creation date in descending order.
+// If creation date is nil, the AMIs will be sorted by name in ascending order.
```

Comment:
This also includes ties as well, right?
```suggestion
// If creation date is nil or two AMIs have the same creation date, the AMIs will be sorted by name in ascending order.
```